sqlmesh/core/engine_adapter/athena.py

```diff
@@ -326,6 +353,15 @@ def _query_table_type(
                 return "hive"
         return "iceberg"
 
+    def _is_hive_partitioned_table(self, table_name: exp.Table) -> bool:
```

Comment:
if this takes a Table, maybe just call this arg table

Reply:
Updated, yeah I kept switching between `TableName` and `exp.Table` because the parent classes use `TableName` but checking for strings all the time was getting tedious.

The general pattern throughout the codebase of making all the methods take a `str` or an `exp.Expression` is nice in some ways but annoying in others